[CI] DeleteProjectActionTests testSimpleDelete failing

**Build Scans:**
- [elasticsearch-periodic #9636 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/meanop3yzm3fk)
- [elasticsearch-periodic #9622 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/mynlnltwnj6o6)

**Reproduction Line:**
```
./gradlew ":test:external-modules:test-multi-project:test" --tests "org.elasticsearch.multiproject.action.DeleteProjectActionTests.testSimpleDelete" -Dtests.seed=C59C794EA2C88846 -Dtests.locale=en-MH -Dtests.timezone=America/Lower_Princes -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.multiproject.action.DeleteProjectActionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSimpleDelete'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[LEAK: resource was not cleaned up before it was garbage-collected.
Recent access records: 
Created at:
	in [TEST-DeleteProjectActionTests.testDeleteNonExisting-seed#[C59C794EA2C88846]][testDeleteNonExisting]
	org.elasticsearch.multiproject.action.DeleteProjectActionTests.testDeleteNonExisting(DeleteProjectActionTests.java:62)
	java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	java.base/java.lang.reflect.Method.invoke(Method.java:580)
	com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)
	com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
	com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementA
[truncated]
```

**Issue Reasons:**
- [main] 2 failures in test testSimpleDelete (2.3% fail rate in 88 executions)
- [main] 2 failures in step openjdk23_checkpart1_java-matrix (66.7% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/external-modules/multi-project/src/test/java/org/elasticsearch/multiproject/action/DeleteProjectActionTests.java
+++ b/test/external-modules/multi-project/src/test/java/org/elasticsearch/multiproject/action/DeleteProjectActionTests.java
@@ -59,12 +59,12 @@ public class DeleteProjectActionTests extends ESTestCase {
         var projects = randomList(1, 5, ESTestCase::randomUniqueProjectId);
         var deletedProjects = randomSubsetOf(projects);
         var state = buildState(projects);
-        var listener = ActionListener.assertAtLeastOnce(
-            ActionTestUtils.<AcknowledgedResponse>assertNoSuccessListener(e -> assertTrue(e instanceof IllegalArgumentException))
-        );
+        var listener = ActionListener.assertAtLeastOnce(ActionTestUtils.<AcknowledgedResponse>assertNoSuccessListener(e -> {
+            assertTrue(e instanceof IllegalArgumentException);
+        }));
         var nonExistingTask = createTask(randomUniqueProjectId(), listener);
         var tasks = Stream.concat(Stream.of(nonExistingTask), deletedProjects.stream().map(this::createTask)).toList();
-        var result = ClusterStateTaskExecutorUtils.executeIgnoringFailures(state, executor, tasks);
+        var result = ClusterStateTaskExecutorUtils.executeHandlingResults(state, executor, tasks, t -> {}, DeleteProjectTask::onFailure);
         for (ProjectId deletedProject : deletedProjects) {
             assertNull(result.metadata().projects().get(deletedProject));
             assertNull(result.globalRoutingTable().routingTables().get(deletedProject));

--- a/test/external-modules/multi-project/src/test/java/org/elasticsearch/multiproject/action/DeleteProjectActionTests.java
+++ b/test/external-modules/multi-project/src/test/java/org/elasticsearch/multiproject/action/DeleteProjectActionTests.java
@@ -59,9 +59,9 @@ public class DeleteProjectActionTests extends ESTestCase {
         var projects = randomList(1, 5, ESTestCase::randomUniqueProjectId);
         var deletedProjects = randomSubsetOf(projects);
         var state = buildState(projects);
-        var listener = ActionListener.assertAtLeastOnce(ActionTestUtils.<AcknowledgedResponse>assertNoSuccessListener(e -> {
-            assertTrue(e instanceof IllegalArgumentException);
-        }));
+        var listener = ActionListener.assertAtLeastOnce(
+            ActionTestUtils.<AcknowledgedResponse>assertNoSuccessListener(e -> assertTrue(e instanceof IllegalArgumentException))
+        );
         var nonExistingTask = createTask(randomUniqueProjectId(), listener);
         var tasks = Stream.concat(Stream.of(nonExistingTask), deletedProjects.stream().map(this::createTask)).toList();
         var result = ClusterStateTaskExecutorUtils.executeHandlingResults(state, executor, tasks, t -> {}, DeleteProjectTask::onFailure);

